index-scheduler/src/error.rs

```diff
@@ -132,6 +132,24 @@ pub enum Error {
     TaskDatabaseUpdate(Box<Self>),
     #[error(transparent)]
     HeedTransaction(heed::Error),
+
+    #[cfg(test)]
+    #[error("Planned failure for tests.")]
+    PlannedFailure,
+}
+
+impl Error {
+    pub fn is_recoverable(&self) -> bool {
+        match self {
+            Error::CorruptedTaskQueue
+            | Error::TaskDatabaseUpdate(_)
+            | Error::HeedTransaction(_)
+            | Error::CreateBatch(_) => false,
+            #[cfg(test)]
+            Error::PlannedFailure => false,
+            _ => true,
```

Comment:
Could you be explicit on these variants? This way adding a variant to the enum will force us to think about this match.

Reply:
Done, I kept the same ordering as in the defining enum but maybe alphabetical ordering would be better in cases one searches a particular value in that long list.